docker-entrypoint.sh

```diff
@@ -292,6 +292,31 @@ docker_setup_db() {
 		EOSQL
 	fi
 
+	local mysqlAtLocalhost=
+	local mysqlAtLocalhostGrants=
+	# Install mysql@localhost user
+	if [ -n "$MARIADB_MYSQL_LOCALHOST_USER" ] || [ -n "$MARIADB_MYSQL_LOCALHOST_GRANTS" ]; then
```

Comment:
`MARIADB_MYSQL_LOCALHOST_GRANTS` shouldn't imply `MARIADB_MYSQL_LOCALHOST_USER`.

```suggestion
	if [ -n "$MARIADB_MYSQL_LOCALHOST_USER" ]; then
```

Reply:
I'm not convinced even after leaving this a few days. A grants implies the user has to exist. I like having the `MARIADB_MYSQL_LOCALHOST_USER=1` to give defaults usage without a user giving up on exact privileges and just using `ALL`.

Conceptually sure having one setting imply another is a tight coupling is to be avoided. I see in this case the underlying concepts are so coupled its going to be annoying to a user that specified `*GRANTS` and misses creating the `*USER`. Its almost like the `MARIADB_{USER,PASSWORD,DATABASE}` trifecta that we have now where all 3 are needed for those to be considered.

Reply:
noted and changed - https://github.com/MariaDB/mariadb-docker/commit/5c9b0ddb4711f6c6f2e96ead6e2d991443cea192